lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -1207,31 +1210,58 @@ impl Deref for Scene {
 }
 
 impl Scene {
+    /// Perform early phase of scene update. This includes updating camera and the layout of all
+    /// display objects. No GPU buffers are updated yet, giving the opportunity to perform
+    /// additional updates that affect the layout of display objects after the main scene layout
+    /// has been performed.
+    ///
+    /// During this phase, the layout updates can be observed using `on_updated` FRP events on each
+    /// individual display object. Any further updates to the scene may require the `update`
+    /// method to be manually called on affected objects in order to affect rendering during
+    /// this frame.
     #[profile(Debug)]
-    // FIXME:
-    #[allow(unused_assignments)]
-    pub fn update(&self, time: animation::TimeInfo) -> UpdateStatus {
-        if let Some(context) = &*self.context.borrow() {
-            debug_span!("Updating.").in_scope(|| {
+    pub fn early_update(&self, time: animation::TimeInfo) -> UpdateStatus {
+        if self.context.borrow().is_some() {
+            debug_span!("Early update.").in_scope(|| {
                 let mut scene_was_dirty = false;
                 self.frp.frame_time_source.emit(time.since_animation_loop_started.unchecked_raw());
                 // Please note that `update_camera` is called first as it may trigger FRP events
                 // which may change display objects layout.
-                scene_was_dirty = self.update_camera(self) || scene_was_dirty;
+                scene_was_dirty |= self.update_camera(self);
```

Comment:
revert pls

Reply:
It does work as I expected it to: [play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c86d4ff2c083c9085cd910db4f43b3c5](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c86d4ff2c083c9085cd910db4f43b3c5)